### ISSUE TITLE
Better matching of type is needed

### ISSUE THREAD

**Issue by wolfgangkarall** [MestreLion/humblebundle#3]
I can't choose to download 'PDF' if there's also 'PDF (HD)':

```
$ ./humblebundle --type 'PDF' -p ebook --download xkcdvolume0 -f /home/humbledownload/downloads/
2015-01-28 14:24:35,297 ERROR           Too many download candidates for 'xkcd: volume 0' [xkcdvolume0]. Improve criteria to narrow it down.
        'PDF'   35 MB   xkcd-volume0-low.pdf
        'PDF (HD)'      132.7 MB        xkcd-volume0-high.pdf
```

**Comment by MestreLion** [comment]
That happens because `--type` match types _starting_ with the given string. Changing that behavior to an exact match would fix it, but I'm not sure if it's the best approach. Perhaps adding an `--exact` or --`type-exact` option? Any ideas?

**Comment by wolfgangkarall** [comment]
Yeah, doesn't have to be exact matching all the time. A --type-exact flag switch would be better I guess.

**Comment by MestreLion** [comment]
Yeah, looking at the source and at `gamedata.json` I realized it's a bit more tricky than I imagined. Current matching is not _starting with_ but _containing_, which is far less strict. And the game data expect that in its `download` fields. Otherwise, a small change in how Humble Bundle labels their download buttons (say, from `Download .deb` to `DEB version` would break things.

Oh the other hand, an extra argument (say `--type-exact`) would make an already quite complex algorithm even move messy, not to mention it seems (well, it _is_) a lazy UI.

A 3rd option would be to successively try more restrictive rules (containing, starting with, is) if (and only if) the user specified a `--type` and there are too many candidates after exhausting all other rules.

While this may be the best option (user and compatibility wise), it is also quite complex and hard to implement, so I'm still open to suggestions .
